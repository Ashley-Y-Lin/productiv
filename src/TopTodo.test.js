import React from "react";
import { render } from "@testing-library/react";
import TopTodo from "./TopTodo";
import { TEST_TODOS } from "./_testCommon";

describe("TopTodo component", function () {
  it("renders without crashing", function () {
    render(
      <TopTodo todos={TEST_TODOS} />
    );
  });

  it("contains expected div section with className", function () {
    const { container, debug } = render(
      <TopTodo todos={TEST_TODOS} />
    );
    const div = container.querySelector("div");
    debug(div);
    expect(div).toHaveClass("TopTodo");
  });

  it("matches snapshot", function () {
    const { container } = render(
      <TopTodo todos={TEST_TODOS} />
    );
    expect(container).toMatchSnapshot();
  });

});
