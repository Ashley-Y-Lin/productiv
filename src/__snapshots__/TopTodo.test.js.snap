// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopTodo component matches snapshot 1`] = `
<div>
  <div
    class="TopTodo"
  >
    <div
      class="Todo"
    >
      <div>
        <b>
          Make dinner
        </b>
         
        <small>
          (priority: 
          1
          )
        </small>
      </div>
      <div>
        <small>
          Cook something healthy
        </small>
      </div>
    </div>
  </div>
</div>
`;
